sqlmesh/dbt/model.py

```diff
@@ -187,16 +177,21 @@ def model_kind(self, target: TargetConfig) -> ModelKind:
                     and strategy not in INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES
                 ):
                     raise ConfigError(
-                        f"{self.sql_name}: SQLMesh IncrementalByUniqueKey is not compatible with '{strategy}'"
+                        f"{self.sql_name}: SQLMesh incremental by unique key strategy is not compatible with '{strategy}'"
                         f" incremental strategy. Supported strategies include {collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES)}."
                     )
                 return IncrementalByUniqueKeyKind(unique_key=self.unique_key, **incremental_kwargs)
 
-            raise ConfigError(
-                f"{self.sql_name}: Incremental materialization requires either a "
-                f"time_column ({collection_to_str(INCREMENTAL_BY_TIME_STRATEGIES)}) or a "
-                f"unique_key ({collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES.union(['none']))}) configuration."
+            logger.warning(
+                "Incremental materialization in model '%s' requires either a time_column (%s) or a unique_key (%s) configuration",
+                self.sql_name,
+                collection_to_str(INCREMENTAL_BY_TIME_STRATEGIES),
+                collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES.union(["none"])),
+            )
+            strategy = self.incremental_strategy or target.default_incremental_strategy(
+                IncrementalUnmanagedKind
             )
+            return IncrementalUnmanagedKind(insert_overwrite=strategy == "insert_overwrite")
```

Comment:
How do we know whether or not the incremental by time or incremental by unique key strategies will work if a time_column or unique_key is specified? If the only where clause is an `is_incremental` block, I could be wrong, but I think we need to fall back to `IncrementalUnmanagedKind`.

Reply:
I don't understand this comment. If there is `unique_key` or `time_column` available then the user's intent is pretty clear.

Reply:
unique_key is a dbt config field. It wasn't necessarily added as part of a sqlmesh migration.

Reply:
unique_key is a dbt config field. It wasn't necessarily added as part of a sqlmesh migration.